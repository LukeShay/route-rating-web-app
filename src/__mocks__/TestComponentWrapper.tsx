/* eslint @typescript-eslint/no-empty-function: 0, @typescript-eslint/explicit-function-return-type: 0 */
import React from "react";
import { IGymsContextState } from "../context/gyms/gymsStore";
import { IViewContextState } from "../context/view/viewStore";
import { IUserContextState } from "../context/user/userStore";
import { StaticRouter } from "react-router";
import GymsStoreMock from "./gymsStoreMock";
import ViewStoreMock from "./viewStoreMock";
import UserStoreMock from "./userStoreMock";

export interface ITestComponentWrapperProps {
  children: React.ReactNode;
  gymsState: IGymsContextState;
  location: string;
  userState: IUserContextState;
  viewState: IViewContextState;
}

const TestComponentWrapper: React.FC<ITestComponentWrapperProps> = ({
  children,
  gymsState,
  location,
  userState,
  viewState
}): JSX.Element => {
  return (
    <StaticRouter location={`https://lukeshay.com/${location}`}>
      <GymsStoreMock state={gymsState} dispatch={() => {}}>
        <ViewStoreMock state={viewState} dispatch={() => {}}>
          <UserStoreMock state={userState} dispatch={() => {}}>
            {children}
          </UserStoreMock>
        </ViewStoreMock>
      </GymsStoreMock>
    </StaticRouter>
  );
};

export default TestComponentWrapper;
