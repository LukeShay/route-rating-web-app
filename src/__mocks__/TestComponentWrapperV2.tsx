/* eslint @typescript-eslint/no-empty-function: 0, @typescript-eslint/explicit-function-return-type: 0 */
import React from "react";
import { IGymsContextState } from "../context/gyms/gymsStore";
import { IUserContextState } from "../context/user/userStore";
import { IViewContextState } from "../context/view/viewStore";
import GymsStoreMock from "./gymsStoreMock";
import ViewStoreMock from "./viewStoreMock";
import UserStoreMock from "./userStoreMock";

export interface ITestComponentWrapperV2Props {
  children: React.ReactNode;
  gymsState: IGymsContextState;
  userState: IUserContextState;
  viewState: IViewContextState;
}

const TestComponentWrapperV2: React.FC<ITestComponentWrapperV2Props> = ({
  children,
  gymsState,
  userState,
  viewState
}): JSX.Element => {
  return (
    <GymsStoreMock state={gymsState} dispatch={() => {}}>
      <ViewStoreMock state={viewState} dispatch={() => {}}>
        <UserStoreMock state={userState} dispatch={() => {}}>
          {children}
        </UserStoreMock>
      </ViewStoreMock>
    </GymsStoreMock>
  );
};

export default TestComponentWrapperV2;
