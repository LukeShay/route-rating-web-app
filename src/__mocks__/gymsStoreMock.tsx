import { GymsContext, IGymsContextState } from '../context/gyms/gymsStore';
import React from 'react';

export interface IGymsStoreMockProps {
  state: IGymsContextState;
  dispatch: any;
}

const GymsStoreMock: React.FC<IGymsStoreMockProps> = ({
  state,
  dispatch,
  children,
}): JSX.Element => (
  <GymsContext.Provider value={{ state, dispatch }}>
    {children}
  </GymsContext.Provider>
);

export default GymsStoreMock;
