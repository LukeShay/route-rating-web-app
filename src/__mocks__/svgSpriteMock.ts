export default {
  id: 'svg-sprite-id',
  viewBox: '0 0 1 1',
  url: '/sprite.svg#svg-sprite-id',
};
