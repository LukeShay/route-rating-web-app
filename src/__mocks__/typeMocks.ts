import { Gym, GymPageType, Route, User, Wall, WallTypes } from '../types';
import { GymsContextState } from '../context/gyms/gymsStore';
import { ViewContextState } from '../context/view/viewStore';
import { UserContextState } from '../context/user/userStore';

export const testRouteOne: Route = {
  averageGrade: 'GRADE_5_9',
  averageRating: 4,
  gymId: 'gym1',
  holdColor: 'Blue',
  id: 'route1',
  name: 'RouteOne',
  setter: 'LukeOne',
  types: [WallTypes.TOP_ROPE, WallTypes.LEAD],
  wallId: 'wall1',
};

export const testRouteTwo: Route = {
  averageGrade: 'GRADE_5_12ab',
  averageRating: 4,
  gymId: 'gym1',
  holdColor: 'Red',
  id: 'route2',
  name: 'RouteTwo',
  setter: 'LukeTwo',
  types: [WallTypes.TOP_ROPE, WallTypes.LEAD],
  wallId: 'wall1',
};

export const testWallOne: Wall = {
  gymId: 'gym1',
  id: 'wall1',
  name: 'Wall1',
  routes: [testRouteOne, testRouteTwo],
  types: [WallTypes.LEAD, WallTypes.TOP_ROPE],
};

export const testWallTwo: Wall = {
  gymId: 'gym1',
  id: 'wall2',
  name: 'Wall2',
  routes: [],
  types: [WallTypes.LEAD, WallTypes.TOP_ROPE],
};

export const testGymOne: Gym = {
  address: '1234 Five Street',
  authorizedEditors: ['id'],
  city: 'Ames',
  email: 'abc@d.com',
  id: 'gym1',
  logoUrl: '',
  name: 'Gym One',
  phoneNumber: '1234567890',
  photoUrl: '',
  state: 'Iowa',
  walls: [testWallOne, testWallTwo],
  website: 'lukeshay.com',
  zipCode: '50014',
};

export const testGymTwo: Gym = {
  address: '1234 Five Street',
  authorizedEditors: ['id'],
  city: 'Ames',
  email: 'abc@d.com',
  id: 'gym2',
  logoUrl: '',
  name: 'Gym Two',
  phoneNumber: '1234567890',
  photoUrl: '',
  state: 'Iowa',
  walls: [testWallOne, testWallTwo],
  website: 'lukeshay.com',
  zipCode: '50014',
};

export const testUserEditor: User = {
  authority: 'ADMIN',
  city: 'Ames',
  country: 'USA',
  email: 'email',
  firstName: 'Name',
  id: 'id',
  lastName: 'Last',
  password: 'password',
  phoneNumber: '1111111111',
  role: 'ADMIN_ROLE',
  session: null,
  state: 'IA',
  username: 'username',
};

export const mockGymsState: GymsContextState = {
  page: { content: [testGymOne, testGymTwo] } as GymPageType,
};

export const darkDesktopState: ViewContextState = {
  mobile: false,
  theme: 'DARK_THEME',
};

export const signedInEditorState: UserContextState = {
  user: testUserEditor,
};

export const signedOutState: UserContextState = {
  user: null,
};
