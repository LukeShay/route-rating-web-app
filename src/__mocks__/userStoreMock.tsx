import { IUserContextState, UserContext } from '../context/user/userStore';
import React from 'react';

export interface IUserStoreMockProps {
  state: IUserContextState;
  dispatch: any;
}

const UserStoreMock: React.FC<IUserStoreMockProps> = ({
  state,
  dispatch,
  children,
}): JSX.Element => (
  <UserContext.Provider value={{ state, dispatch }}>
    {children}
  </UserContext.Provider>
);

export default UserStoreMock;
