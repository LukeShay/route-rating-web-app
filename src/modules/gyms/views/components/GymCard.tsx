import {
  Card,
  CardContent,
  CardMedia,
  Typography,
  createStyles,
  makeStyles
} from "@material-ui/core";
import React from "react";
import { Gym } from "../../../../types";

const useStyles = makeStyles(() =>
  createStyles({
    card: {
      borderRadius: "10px",
      display: "flex"
    },
    cardWrapper: {
      alignItems: "center",
      display: "flex",
      justifyContent: "center",
      paddingBottom: "10px",
      paddingTop: "10px",
      width: "100%"
    },
    information: {
      paddingBottom: "5px",
      paddingLeft: "10px",
      paddingTop: "5px"
    },
    photo: {
      borderRadius: "10px",
      height: "96%"
    },
    photoWrapper: {
      alignItems: "center",
      display: "flex",
      justifyContent: "center",
      width: "50%"
    }
  })
);

const useMobileStyles = makeStyles(() =>
  createStyles({
    card: {
      borderRadius: "5px"
    },
    cardWrapper: {
      paddingBottom: "10px",
      paddingTop: "10px",
      width: "100%"
    },
    photo: {
      borderRadius: "10px",
      height: "96%"
    },
    photoWrapper: {
      alignItems: "center",
      borderRadius: "10px",
      display: "flex",
      justifyContent: "center",
      width: "100%"
    }
  })
);

interface IGymCardVariantsProps {
  cardClass?: string;
  gym: Gym;
  onClick?(even: any): void;
}

const GymCardDesktop: React.FC<IGymCardVariantsProps> = ({
  cardClass,
  gym,
  onClick
}): JSX.Element => {
  const classes = useStyles();

  return (
    <div className={classes.cardWrapper} onClick={onClick}>
      <Card className={cardClass || classes.card} data-test-id="gym-card-test-id">
        <CardMedia className={classes.photoWrapper}>
          <img
            src={"https://" + gym.photoUrl}
            alt="This gym does not have a photo."
            className={classes.photo}
          />
        </CardMedia>
        <CardContent>
          <Typography variant="h4">{gym.name}</Typography>
          <div className={classes.information}>
            <Typography variant="body1">{gym.website}</Typography>
            <Typography variant="body1">
              {gym.address}
              <br />
              {gym.city + ", " + gym.state + " " + gym.zipCode}
            </Typography>
            <Typography variant="body1">{gym.email}</Typography>
            <Typography variant="body1">{gym.phoneNumber}</Typography>
          </div>
        </CardContent>
      </Card>
    </div>
  );
};

const GymCardMobile: React.FC<IGymCardVariantsProps> = ({
  gym,
  onClick
}): JSX.Element => {
  const classes = useMobileStyles();

  return (
    <div className={classes.cardWrapper} onClick={onClick}>
      <Card className={classes.card} data-test-id="gym-card-test-id">
        <CardMedia
          className={classes.photoWrapper}
          style={{
            height: `${window.innerWidth * (2 / 3) - 10}px`
          }}
        >
          <img
            className={classes.photo}
            src={"https://" + gym.photoUrl}
            alt="This gym does not have a photo."
          />
        </CardMedia>
        <CardContent>
          <Typography variant="h4">{gym.name}</Typography>
          <div>
            <Typography variant="body1">{gym.website}</Typography>
            <Typography variant="body1">
              {gym.address}
              <br />
              {gym.city + ", " + gym.state + " " + gym.zipCode}
            </Typography>
            <Typography variant="body1">{gym.email}</Typography>
            <Typography variant="body1">{gym.phoneNumber}</Typography>
          </div>
        </CardContent>
      </Card>
    </div>
  );
};

export interface IGymCardProps {
  desktopCardClass?: string;
  gym: Gym;
  mobile: boolean;
  mobileCardClass?: string;
  onClick?(even: any): void;
}

const GymCard: React.FC<IGymCardProps> = ({
  desktopCardClass,
  gym,
  mobile,
  mobileCardClass,
  onClick
}) => {
  return mobile ? (
    <GymCardMobile gym={gym} onClick={onClick} cardClass={mobileCardClass} />
  ) : (
    <GymCardDesktop gym={gym} onClick={onClick} cardClass={desktopCardClass} />
  );
};

export default GymCard;
