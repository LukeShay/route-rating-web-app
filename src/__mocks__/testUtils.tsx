import { render, RenderResult } from '@testing-library/react';
import React from 'react';
import TestComponentWrapperV2 from './TestComponentWrapperV2';
import * as TypeMocks from './typeMocks';

interface WrapperProps {
  children: React.ReactNode;
}

const SignedInDesktop: React.FC<WrapperProps> = ({ children }): JSX.Element => {
  return (
    <TestComponentWrapperV2
      gymsState={TypeMocks.mockGymsState}
      userState={TypeMocks.signedInEditorState}
      viewState={TypeMocks.darkDesktopState}
    >
      {children}
    </TestComponentWrapperV2>
  );
};

export const renderSignedInDesktop = (ui, options): RenderResult =>
  render(ui, { wrapper: SignedInDesktop, ...options });

const SignedOutDesktop: React.FC<WrapperProps> = ({
  children,
}): JSX.Element => {
  return (
    <TestComponentWrapperV2
      gymsState={TypeMocks.mockGymsState}
      userState={TypeMocks.signedOutState}
      viewState={TypeMocks.darkDesktopState}
    >
      {children}
    </TestComponentWrapperV2>
  );
};

export const renderSignedOutDesktop = (ui, options): RenderResult =>
  render(ui, { wrapper: SignedOutDesktop, ...options });

export * from '@testing-library/react';
