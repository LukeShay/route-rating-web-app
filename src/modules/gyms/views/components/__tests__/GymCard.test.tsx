import React from "react";
import GymCard from "../GymCard";
import { shallow } from "../../../../../../configs/setupEnzyme";
import { shallowToJson } from "enzyme-to-json";
import * as TypeMocks from "../../../../../__mocks__/typeMocks";

describe("GymCard", () => {
  it("display correctly for desktop.", () => {
    const gymsPage = shallow(
      <GymCard gym={TypeMocks.testGymOne} mobile={false} />
    );
    expect(shallowToJson(gymsPage)).toMatchSnapshot();
  });

  it("display correctly for mobile.", () => {
    const gymsPage = shallow(
      <GymCard gym={TypeMocks.testGymOne} mobile={true} />
    );
    expect(shallowToJson(gymsPage)).toMatchSnapshot();
  });
});
