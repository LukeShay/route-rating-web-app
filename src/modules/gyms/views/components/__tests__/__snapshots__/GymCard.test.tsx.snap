// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GymCard display correctly for desktop. 1`] = `
<GymCardDesktop
  gym={
    Object {
      "address": "1234 Five Street",
      "authorizedEditors": Array [
        "id",
      ],
      "city": "Ames",
      "email": "abc@d.com",
      "id": "gym1",
      "logoUrl": "",
      "name": "Gym One",
      "phoneNumber": "1234567890",
      "photoUrl": "",
      "state": "Iowa",
      "walls": Array [
        Object {
          "gymId": "gym1",
          "id": "wall1",
          "name": "Wall1",
          "routes": Array [
            Object {
              "averageGrade": "GRADE_5_9",
              "averageRating": 4,
              "gymId": "gym1",
              "holdColor": "Blue",
              "id": "route1",
              "name": "RouteOne",
              "setter": "LukeOne",
              "types": Array [
                "TOP_ROPE",
                "LEAD",
              ],
              "wallId": "wall1",
            },
            Object {
              "averageGrade": "GRADE_5_12ab",
              "averageRating": 4,
              "gymId": "gym1",
              "holdColor": "Red",
              "id": "route2",
              "name": "RouteTwo",
              "setter": "LukeTwo",
              "types": Array [
                "TOP_ROPE",
                "LEAD",
              ],
              "wallId": "wall1",
            },
          ],
          "types": Array [
            "LEAD",
            "TOP_ROPE",
          ],
        },
        Object {
          "gymId": "gym1",
          "id": "wall2",
          "name": "Wall2",
          "routes": Array [],
          "types": Array [
            "LEAD",
            "TOP_ROPE",
          ],
        },
      ],
      "website": "lukeshay.com",
      "zipCode": "50014",
    }
  }
/>
`;

exports[`GymCard display correctly for mobile. 1`] = `
<GymCardMobile
  gym={
    Object {
      "address": "1234 Five Street",
      "authorizedEditors": Array [
        "id",
      ],
      "city": "Ames",
      "email": "abc@d.com",
      "id": "gym1",
      "logoUrl": "",
      "name": "Gym One",
      "phoneNumber": "1234567890",
      "photoUrl": "",
      "state": "Iowa",
      "walls": Array [
        Object {
          "gymId": "gym1",
          "id": "wall1",
          "name": "Wall1",
          "routes": Array [
            Object {
              "averageGrade": "GRADE_5_9",
              "averageRating": 4,
              "gymId": "gym1",
              "holdColor": "Blue",
              "id": "route1",
              "name": "RouteOne",
              "setter": "LukeOne",
              "types": Array [
                "TOP_ROPE",
                "LEAD",
              ],
              "wallId": "wall1",
            },
            Object {
              "averageGrade": "GRADE_5_12ab",
              "averageRating": 4,
              "gymId": "gym1",
              "holdColor": "Red",
              "id": "route2",
              "name": "RouteTwo",
              "setter": "LukeTwo",
              "types": Array [
                "TOP_ROPE",
                "LEAD",
              ],
              "wallId": "wall1",
            },
          ],
          "types": Array [
            "LEAD",
            "TOP_ROPE",
          ],
        },
        Object {
          "gymId": "gym1",
          "id": "wall2",
          "name": "Wall2",
          "routes": Array [],
          "types": Array [
            "LEAD",
            "TOP_ROPE",
          ],
        },
      ],
      "website": "lukeshay.com",
      "zipCode": "50014",
    }
  }
/>
`;
